src/internal/components/QrCode/QrCodeSvg.tsx

```diff
@@ -0,0 +1,183 @@
+import { useMemo } from 'react';
+import { useOnchainKit } from '../../../core-react/useOnchainKit';
+import {
+  GRADIENT_END_COORDINATES,
+  GRADIENT_START_COORDINATES,
+  QR_CODE_SIZE,
+  QR_LOGO_BACKGROUND_COLOR,
+  QR_LOGO_RADIUS,
+  QR_LOGO_SIZE,
+  linearGradientStops,
+  ockThemeToLinearGradientColorMap,
+  ockThemeToRadiamGradientColorMap,
+  presetGradients,
+} from './gradientConstants';
+import { useCorners } from './useCorners';
+import { useDotsPath } from './useDotsPath';
+import { useLogo } from './useLogo';
+import { useMatrix } from './useMatrix';
+
+function coordinateAsPercentage(coordinate: number) {
+  return `${coordinate * 100}%`;
+}
+
+export type QRCodeSVGProps = {
+  value: string;
+  size?: number;
+  backgroundColor?: string;
+  logo?: React.ReactNode;
+  logoSize?: number;
+  logoBackgroundColor?: string;
+  logoMargin?: number;
+  logoBorderRadius?: number;
+  quietZone?: number;
+  quietZoneBorderRadius?: number;
+  ecl?: 'L' | 'M' | 'Q' | 'H';
+  gradientType?: 'radial' | 'linear';
```

Comment:
I know many of these are optional but curious if we even need all of these props - do we want this low level of customization for a QR code? I imagine that a few of these params will always stay the same and therefore we can just default in the func.